lib/IntermediateRepresentation/program.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FORTRAN_INTERMEDIATEREPRESENTATION_PROGRAM_H_
+#define FORTRAN_INTERMEDIATEREPRESENTATION_PROGRAM_H_
+
+#include "common.h"
+#include "procedure.h"
```

Comment:
These headers are mutually dependent.  You might need some forward declarations of classes.  (Ignore this note if the C++ implementation files are already building fine.)

EDIT: Never mind, I see some forward declarations later.  But maybe you don't need the `#include` (?)